packages/expo/src/logs/LogSerialization.ts

```diff
@@ -73,7 +73,16 @@ function _stringifyLogData(data: unknown[]): string[] {
     if (typeof item === 'string') {
       return item;
     } else {
-      return prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      let result = prettyFormat(item, { plugins: [ReactNodeFormatter] });
+      // check the size of string returned
+      if (result.length > 10000) {
+        let truncatedResult = result.substring(0, 10000);
+        // truncate the result string to the first 10000 characters
+        truncatedResult += '...';
```

Comment:
add the message explaining what is going on: https://github.com/expo/expo/pull/4419#discussion_r290526911